### PR TITLE
CLR-13, CLR-14: Update code to fix bug that causes crash on non-compliance event

### PR DESCRIPTION
**Pull request by avirmani-miso**
1. ROI updated to get the sink in the location
2. Fixed the non-compliance event bug that causes the system crash

### CODE REVIEW

**Review by dpalacios-miso — APPROVED** [review]
I'm just going to rubber stamp these for now.  We'll start tracking real development in the next couple of weeks.

File: image_processing/contour_filter.py [hunk 1]
```diff
@@ -20,7 +20,10 @@ def filter_contours_based_on_area(
         countour_areas = list()  ## only for debugging
         for contour in contours:
             countour_area = cv2.contourArea(contour)
-            if countour_area > 2000:
+            contour_percent = 100*countour_area/(binary_image.shape[0]*binary_image.shape[1])
+            # print("contour_area: ", countour_area)
```
Comment (gchantzialexiou) [inline image_processing/contour_filter.py:24]: please remove this

File: image_processing/contour_filter.py [hunk 2]
```diff
@@ -20,7 +20,10 @@ def filter_contours_based_on_area(
...
+            print("contour_area (%): ", contour_percent)
```
Comment (gchantzialexiou) [inline image_processing/contour_filter.py:25]: please use logging and not print

File: image_processing/video_processor.py [hunk 3]
```diff
@@ -50,10 +50,12 @@ def __init__(
     def process(self, frame: np.ndarray) -> None:
         self._state.handle(self)
         binaryImage = self.background_strategy.estimate(frame)
+        cv2.imshow("binary", binaryImage)
```
Comment (gchantzialexiou) [inline image_processing/video_processor.py:53]: Do we need to show this? Is this needed for the demo? It takes a lot of cpu to imshow images on the jetson

File: main.py [hunk 4]
```diff
@@ -59,9 +60,11 @@
 
         if ret:
             timestamp = cap.get(cv2.CAP_PROP_POS_MSEC)
-
+            dim = (704, 576)
+            frame = cv2.resize(frame, dim, interpolation = cv2.INTER_AREA)
+            cv2.imshow("orig", frame)
```
Comment (gchantzialexiou) [inline main.py:65]: same here. do we need to imshow this for the app?

File: main.py [hunk 5]
```diff
@@ -59,9 +60,11 @@
 
         if ret:
             timestamp = cap.get(cv2.CAP_PROP_POS_MSEC)
-
+            dim = (704, 576)
+            frame = cv2.resize(frame, dim, interpolation = cv2.INTER_AREA)
```
Comment (gchantzialexiou) [inline main.py:64]: why do we resize the image?

File: image_loader.py [hunk 6]
```diff
@@ -27,7 +27,7 @@ def load_images(self):
         self.images["handwashing_compliant"] = cv2.imread(
             "pre_computed_images/handwashing_compliant.png"
         )
-        self.images["handwashing_non_cmpliant"] = cv2.imread(
+        self.images["handwashing_non_compliant"] = cv2.imread(
```
Comment (gchantzialexiou) [inline image_loader.py:30]: good eyes

File: image_processing/contour_filter.py [hunk 7]
```diff
@@ -20,7 +20,10 @@ def filter_contours_based_on_area(
         countour_areas = list()  ## only for debugging
         for contour in contours:
             countour_area = cv2.contourArea(contour)
-            if countour_area > 2000:
+            contour_percent = 100*countour_area/(binary_image.shape[0]*binary_image.shape[1])
```
Comment (gchantzialexiou) [inline image_processing/contour_filter.py:23]: why is this needed?

File: image_processing/contour_filter.py (hunk 1 above)
Comment (avirmani-miso) [reply image_processing/contour_filter.py:24]: Sure, I was experimenting with this, since using exact pixel count may not be a good idea, should the resolution or camera change. I noticed camera feed sent via George had different dimensions than the cv2 videocapture.
I will remove it though.

File: image_processing/contour_filter.py (hunk 2 above)
Comment (avirmani-miso) [reply image_processing/contour_filter.py:25]: okay.

File: image_processing/video_processor.py (hunk 3 above)
Comment (avirmani-miso) [reply image_processing/video_processor.py:53]: I used imshow, since this is what was originally used in your code as well and UI is not ready at the moment. I can remove this once we dockerize the application and UI is a separate component. 
Jetson resources are well within limits (<2GB/ 8GB) for the current code, so we should be good.

File: main.py (hunk 4 above)
Comment (avirmani-miso) [reply main.py:65]: same as previous comment.

File: main.py (hunk 5 above)
Comment (avirmani-miso) [reply main.py:64]: As I mentioned in the previous comment, video recorded directly from camera uses a higher resolution, but videocapture uses lower resolution by default.

Comment (avirmani-miso) [reply main.py:64]: I just checked we can update the default settings for videocapture, I'll add them. This ensures, the roi corner points do not change when processing video or camera feed.

File: image_processing/contour_filter.py (hunk 7 above)
Comment (avirmani-miso) [reply image_processing/contour_filter.py:23]: I'll remove it, it was for an experiment.

**Comment by avirmani-miso** [comment]
Rebased with changes in develop. Need addtional review.

**Review by dpalacios-miso — APPROVED** [review]
